src/index.ts

```diff
@@ -199,6 +208,46 @@ const LICENSE_ENV_VARIABLES = [
   "ACCEPT_MAX_GAS_PRICE_GWEI_VALUE",
 ];
 
+async function pollPooledMinerResults() {
+  const provider = getProvider();
+  const wallet = new Wallet(process.env.PRIVATE_KEY, provider);
+  const submittedNonces = [];
+  while (true) {
+    let result = await axios.get<PooledMinerResult>(process.env.POOLED_MINER_RESULT_URL);
+    let last = result.data;
+    console.log(`polling pooled miner results for address ${wallet.address}, last mined address is ${last.address}...`)
+    if (last.address === wallet.address &&
+      last.error === null &&
+      last.success === true &&
+      submittedNonces.indexOf(last.nonce) == -1) {
+      console.log('found a valid nonce, will mint it if gas price is low enough');
+      const nonce = BigNumber.from(last.nonce);
+
+      const gasStatus = await checkIfGasTooHigh({
+        provider,
+        maxGasGwei: process.env.MAX_GAS_PRICE_GWEI,
+      });
+
+      if (gasStatus == GAS_STATUS.GAS_TOO_HIGH) {
+        console.log(
+          `Gas price is higher than configured MAX_GAS_PRICE_GWEI of ${process.env.MAX_GAS_PRICE_GWEI}, waiting to submit...`
+        );
+      } else {
+        submittedNonces.push(last.nonce);        
+        console.log('will send transaction to mint a new mpunk!')
+        const tx = await mint({ nonce, wallet });
+        console.log(`Nonce submission transaction hash: ${tx.hash}`);
+      }
+
+    }
+    await sleep(5000);
+  }
+}
+
+if (process.env.POLL_POOLED_MINER_RESULTS == 'true') {
+  pollPooledMinerResults();
+}
```

Comment:
Let's put this within the `app.listen` `try { } catch { }` block!

Reply:
Yup, way better.